### ISSUE TITLE
ReflectionObject adapter does not override getReflectionConstants

### ISSUE THREAD

**Issue by michaeljoelphillips** [Roave/BetterReflection#715]
See the failing test case at 0f50ad73cbf2dff7e7252864680e16ce20603e1e and the [thrown exception in core](https://github.com/php/php-src/blob/php-7.4.13/ext/reflection/php_reflection.c#L98):
```php

            ['getConstant', null, 'a', ['foo']],
            ['getConstants', null, ['a', 'b'], []],
+           ['getReflectionConstants', null, ['a', 'b'], []],
            ['getInterfaces', null, [$mockClassLike], []],
```
```
1) Roave\BetterReflectionTest\Reflection\Adapter\ReflectionObjectTest::testAdapterMethods with data set #20 ('getReflectionConstants', null, array('a', 'b'), array())
Error: Internal error: Failed to retrieve the reflection object

/home/nomad/Code/BetterReflection/test/unit/Reflection/Adapter/ReflectionObjectTest.php:128
```

**Comment by Ocramius** [comment]
@michaeljoelphillips unclear what has to be done here - I get that core throws, but what are we doing wrong on our end?

**Comment by michaeljoelphillips** [comment]
See the fix in #721.  Since we don't call the parent constructor in the adapter classes, we can't allow the parent reflection methods to be called.
